ProjFS.Mac/PrjFSKext/PrjFSKext/KauthHandler.cpp

```diff
@@ -723,6 +726,17 @@ static bool ShouldHandleVnodeOpEvent(
 
     *kauthResult = KAUTH_RESULT_DEFER;
     
+    if (ActionBitIsSet(action, KAUTH_VNODE_ACCESS))
```

Comment:
Should we take a PerfSample here? I think it'd be interesting to see how often we're hitting this codepath during a build and to see how expensive this test is in comparison to the other tests in ShouldHandleVnodeOpEvent.

Reply:
Sure I can add a PerfSample.  Initially I didn't think it'd be interesting but you bring up a good point that it might be interesting to see how frequently we're hitting this code path.

Reply:
Sample output with change:

```
   Counter                             [ Samples  ][Total time (ns)][Mean (ns)   ][Min (ns)][Max (ns)  ][|←1.00ns  |←1.02µs  |←1.05ms  |←1.07s   |←1100s       Histogram ]
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 0 HandleVnodeOperation                [    207880][      378107921][      1818][       333][  79312254][        ▁█▅▁▁▁▁▁          ▁                                     ]
 1  |--GetPath                         [    207880][       45231186][       217][        71][     60797][      ▆█▆▁▁▁▁▁▁▁                                                ]
 2  |--ShouldHandleVnodeOpEvent        [    207880][      128725892][       619][        76][     53527][      ▁▁█▄▃▁▁▁▁▁                                                ]
 3  |  |--IsVnodeAccessCheck           [    207880][        7065527][        33][        11][     52352][   ▄█▃▄▁▁ ▁▁▁▁▁▁                                                ]
 4  |  |  |--IgnoredVnodeAccessCheck   [       151]
 5  |  |--VnodeIsOnAllowedFilesystem   [    207729][       10559373][        50][        19][     22117][    █▆▅▁▁▁▁▁▁▁▁                                                 ]
 6  |  |--ShouldIgnoreVnodeType        [    207713][        6130344][        29][        15][     18291][   ▁█▄▁▁▁ ▁▁▁▁▁                                                 ]
 7  |  |  |--Ignored                   [         3]
 8  |  |--TryReadVNodeFileFlags        [    207710][       38705479][       186][        73][     51495][      █▄▅▁▁▁▁▁▁▁                                                ]
 9  |  |  |--NotInAnyRoot              [    207189]
10  |  |--IsFileSystemCrawler          [        49][           4771][        97][        34][       302][     ▆█▂▁                                                       ]
11  |     |--Denied                    [         7]
12  |--TryGetVirtualizationRoot        [        39][         371119][      9515][       390][     39431][        ▂▄ ▁██▄▁                                                ]
13  |  |--FindForVnode                 [        39][         349449][      8960][       223][     38582][       ▂▄  ▂██▄▁                                                ]
14  |  |  |--inner_loop_iterations     [       112][         335230][      2993][        52][     12194][     ▂▆▁  ▁█▅▂                                                  ]
15  |  |--TemporaryDirectory           [         0]
16  |  |--NoRootFound                  [         0]
17  |  |--ProviderOffline              [         6]
18  |  |--CompareProviderPid           [        33][           4988][       151][        58][       336][     ▁▄█▁                                                       ]
19  |     |--OriginatedByProvider      [        31]
20  |--RaisePreDeleteEvent             [         0]
21  |--RaiseEnumerateDirectoryEvent    [         1][       77387355][  77387355][  77387355][  77387355][                          █                                     ]
22  |--RaiseRecursivelyEnumerateEvent  [         0]
23  |--RaiseHydrateFileEvent           [         1][       79278871][  79278871][  79278871][  79278871][                          █                                     ]
24 HandleFileOpOperation               [      7537][      106107168][     14078][       133][    415994][       █▇▄▁▁▁▁▅▅▁▁▁                                             ]
25  |--ShouldHandleFileOpEvent         [      2536][      101957134][     40203][        23][    413594][    ▁▁▁▁     ▁██▁▁▁                                             ]
26  |  |--FindVirtualizationRoot       [      2527][      101242001][     40064][     12442][    413195][             ▁██▁▁▁                                             ]
27  |  |  |--FindForVnode              [      2527][      100150901][     39632][     12169][    412310][             ▁██▁▁▁                                             ]
28  |  |  |  |--inner_loop_iterations  [     19544][       98308281][      5030][       105][    120493][      ▁   ▁█▆▂▁▁▁                                               ]
29  |  |  |--NoRootFound               [      2525]
30  |  |--CompareProviderPid           [         2][            283][       141][       127][       156][      ██                                                        ]
31  |     |--OriginatedByProvider      [         2]
32  |--RaiseRenamedEvent               [         0]
33  |--RaiseHardLinkCreatedEvent       [         0]
34  |--RaiseFileModifiedEvent          [         0]
35  |--RaiseFileCreatedEvent           [         0]
```